### PR TITLE
Add a FastAPI app that exposes ClusterProcessProxy methods RESTfully

### PR DESCRIPTION
**Pull request by ig248**

### CODE REVIEW

File: dask_remote/runner/api.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            n = 0
+        app.cluster._adaptive_stop()
+        app.cluster.scale(n)
+        return MessageResponse(message=f"Scaling to {n} workers")
+
+    @app.post("/adapt", response_model=MessageResponse)
+    async def adapt(minimum: int = 0, maximum: int = None):
+        """Set cluster to adaptive scaling mode."""
+        if maximum is None:
+            maximum = math.inf
```
Comment (capitancambio) [inline dask_remote/runner/api.py:54]: hur

Comment (capitancambio) [reply dask_remote/runner/api.py:54]: btw why don't you set it to math.inf directly? it's not an object

File: poetry.lock [hunk 2]
```diff
@@ -139,6 +139,18 @@ optional = false
 python-versions = ">=2.7"
 version = "0.3"
 
+[[package]]
```
Comment (capitancambio) [inline poetry.lock:142]: maybe hide this guy in the git attributes file?

**Review by capitancambio — APPROVED** [review]

File: dask_remote/runner/api.py (hunk 1 above)
Comment (ig248) [reply dask_remote/runner/api.py:54]: @capitancambio not sure what you mean?

File: dask_remote/runner/api.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    message: str
+
+
+def cluster_api(cluster_proxy: ClusterProcessProxy, fastapi_kwargs: dict):
+    """Create a FastAPI app that exposes given ClusterProcessProxy."""
+    app = FastAPI(**fastapi_kwargs)
+    app.cluster = cluster_proxy
+
+    @app.get("/", response_model=MessageResponse)
+    async def root():
```
Comment (pieeri) [inline dask_remote/runner/api.py:23]: @ig248 maybe a permanent redirect to the API swagger?

File: dask_remote/runner/api.py (hunk 1 above)
Comment (ig248) [reply dask_remote/runner/api.py:54]: :heavy_check_mark:

File: dask_remote/runner/api.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        self, cluster_proxy: ClusterProcessProxy, fastapi_kwargs: dict, uvicorn_kwargs: dict
+    ):
+        self.cluster_proxy = cluster_proxy
+        self.fastapi_kwargs = fastapi_kwargs
+        self.uvicorn_kwargs = uvicorn_kwargs
+        super().__init__()
+
+    @property
+    def app(self):
+        return cluster_api(self.cluster_proxy, fastapi_kwargs=self.fastapi_kwargs)
```
Comment (pieeri) [inline dask_remote/runner/api.py:72]: Neat!
👍

File: dask_remote/runner/api.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+from pydantic import BaseModel
+
+from .cluster_process import ClusterProcessProxy
+
+
+class MessageResponse(BaseModel):
+    message: str
+
+
+def cluster_api(cluster_proxy: ClusterProcessProxy, fastapi_kwargs: dict):
```
Comment (pieeri) [inline dask_remote/runner/api.py]: Would it be possible to type the function result?

**Review by pieeri — APPROVED** [review]
It is a matter of taste ; however, I would have found the PR easier to read with "typed" function results.

**Pull request merged**